basics/uri.adoc

```diff
@@ -25,404 +26,522 @@ specific language governing permissions and limitations
 
-* *MUST*  return `UStatus::Failed` if  `validate(UUri)` or `validateRpcMethod(UUri)` fails or `isRpcResponse()` is false
 
-| `isLocal(UUri): boolean`
-| Return true if: `!UUri.hasAuthority()`
+=== Authority
 
-| `isLongForm(UUri): boolean`
-a| Returns true if the Uri part contains names so that it can be serialized to <<Long URIs>>.
+An authority represents the deployment location of a specific software entity. The location is represented by means of a logical identifier like a domain name (e.g. `mcu1.example.com`) or a Vehicle Identification Number (VIN).
 
-| `isEmpty(UUri): boolean`
-| Returns true if the object contains no information
 
-| `isResolved(UUri): boolean`
-| Returns true if the object contains both names and numbers (ids) such that it can be serialize to both <<Long URIs>> and <<Micro URIs>>
+[.specitem,oft-sid="dsn~authority-name-length~1",oft-needs="impl,utest"]
+--
+A UUri's `authority_name` *MUST NOT* exceed 128 characters in length.
 
-| `isMicroForm(UUri): boolean`
-| Returns true if the Uri part contains the required ids to serialized to Micro format and the fields of the Uri can fit within the specified number of bits and bytes. Both sets of details can be obtained under <<Micro URIs>>
+[source,ocl]
+----
+inv: authority_name.length() <= 128
+----
+--
 
-| `isShortForm(UUri): boolean`
-| Returns true if the Uri contains the required ids to be serialized to <<Short URIs>>.
+=== uEntity
 
-| `isRpcMethod(UResource): boolean`
-| Returns true if: `id < 0x8000 && name == rpc`
+A uProtocol software entity (uEntity) is a piece of software deployed somewhere on a network host.
+uEntities are uniquely identified within a system by means of the type and version of the service interface that they implement and a numerical identifier which denotes the particular instance of the service (type) in case multiple instances are available on the system.
 
-| `isRpcResponse(UResource): boolean`
-| Returns true if: `id==0  && name==rpc && instance==response`
+NOTE: uEntities which produce events for consumption by other uEntities assume a _Service_ role. +
+uEntities which consume events produced by other uEntities assume an _Application_ role. +
+uEntities *MAY* assume both the Service and Application roles.
 
-|===
+[.specitem,oft-sid="dsn~entity-id~1"]
+--
+A UUri's `ue_id` property value determines the type and instance of service being referred to. +
+The value's _most_ significant 16 bits represent the _service instance_. +
+The value's _least_ significant 16 bits represent the _service type_.
```

Comment:
identifier and not type

Reply:
We usually talk about _instances_ of a certain _type_, don't we? A service _identifier_ to me suggests that we are talking about a particular instance already, instead of its type.